numpy/core/shape_base.py

```diff
@@ -240,12 +236,13 @@ def hstack(tup):
     """
     Stack arrays in sequence horizontally (column wise).
 
-    Take a sequence of arrays and stack them horizontally to make
-    a single array. Rebuild arrays divided by `hsplit`.
+    This is equivalent to concatenation along the second dimension for 2d and
+    3d arrays, or along the first dimension for 1d arrays. Rebuilds arrays
```

Comment:
For number of dimensions > 2 it always stacks along second dimension. This seems to imply that it only works for arrays ndim <= 3.